test/RequiredPolyfill.cs

```diff
@@ -0,0 +1,48 @@
+﻿#pragma warning disable SA1636, SA1028 // File header copyright text should match, Code shopuld not contain trailing whitespace
```

Comment:
On the licensing note I'd actually prefer that we copy any shims we need directly from the .NET Source and preserve the MS license, that's what we've done where we have previously copied in shim classes elsewhere I think. Any shim libraries would be based on those anyway, and they're the formal 'specification' of the attribute.

As @tillig mentioned, split the attributes into different files, probably under a 'shims' folder or something?

E.g.
https://github.com/dotnet/runtime/blob/c1ab6c5b2524880ed9368841a0a5d8ead78ea09d/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RequiredMemberAttribute.cs

Reply:
I mean, yeah, [SetsRequiredMembersAttribute](https://github.com/dotnet/runtime/blob/c1ab6c5b2524880ed9368841a0a5d8ead78ea09d/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/SetsRequiredMembersAttribute.cs) is internal too, sadly [CompilerFeatureRequiredAttribute](https://github.com/dotnet/runtime/blob/c1ab6c5b2524880ed9368841a0a5d8ead78ea09d/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CompilerFeatureRequiredAttribute.cs) isn't.

Reply:
So... you'd want a folder structure like
```
..
    shims
        required
            `License.md` "Files in this folder are slightly tweaked copies from https://github.com/dotnet/runtime and are licensed as ..."
            CompilerFeatureRequiredAttribute.cs
            RequiredMemberAttribute.cs
            SetsRequiredMembersAttribute.cs
```
?